cmd/soroban-rpc/internal/db/ledgerentry.go

```diff
@@ -341,21 +341,29 @@ func NewLedgerEntryReader(db *DB) LedgerEntryReader {
 }
 
 func (r ledgerEntryReader) GetLatestLedgerSequence(ctx context.Context) (uint32, error) {
-	return getLatestLedgerSequence(ctx, r.db, r.db.cache)
+	return getLatestLedgerSequence(ctx, NewLedgerReader(r.db), r.db.cache)
+}
+
+func (r ledgerEntryReader) newBaseTx(ctx context.Context) (db.SessionInterface, error) {
+	txSession := r.db.Clone()
+	r.db.cache.RLock()
+	defer r.db.cache.RUnlock()
```

Comment:
the lock is released right after `newBaseTx()` returns, which means that the lock is not protecting anything in NewCachedTx() or NewTx()